internal/adc/translator/apisixroute.go

```diff
@@ -328,6 +329,26 @@ func (t *Translator) buildService(ar *apiv2.ApisixRoute, rule apiv2.ApisixRouteH
 	return service
 }
 
+func getPortFromService(svc *v1.Service, backendSvcPort intstr.IntOrString) (int32, error) {
+	var port int32
+	if backendSvcPort.Type == intstr.Int {
+		port = int32(backendSvcPort.IntValue())
+	} else {
+		found := false
+		for _, servicePort := range svc.Spec.Ports {
+			if servicePort.Name == backendSvcPort.StrVal {
+				port = servicePort.Port
```

Comment:
The `getPortFromService` directly uses `port` instead of `targetPort`.
When using the endpoint as the resolution granularity, this should not be the expected behavior?

Reply:
explained here - https://github.com/apache/apisix-ingress-controller/pull/2553/files#r2347360896